src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py

```diff
@@ -8,512 +8,417 @@
-                                callback.append(tmp)
-                            total = result["TotalRecords"]
-                            timestamp = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M [UTC]")
-                            if "Extras" in result:
-                                extra = result["Extras"]
-                                callback = {
-                                    "success": "true",
-                                    "result": callback,
-                                    "total": total,
-                                    "extra": extra,
-                                    "date": timestamp,
-                                }
-                            else:
-                                callback = {"success": "true", "result": callback, "total": total, "date": timestamp}
-                        else:
-                            callback = {"success": "false", "result": "", "error": "There are no data to display"}
-                    else:
-                        callback = {"success": "false", "result": "", "error": "ParameterNames field is undefined"}
-                else:
-                    callback = {"success": "false", "result": "", "error": "Data structure is corrupted"}
-            else:
-                callback = {"success": "false", "result": "", "error": "There were no data matching your selection"}
-        self.finish(callback)
+            return {"success": "false", "error": result["Message"]}
+
+        data = result["Value"]
+        if "TotalRecords" in data and (total := data["TotalRecords"]) > 0:
+            if "ParameterNames" in data and "Records" in data:
+                if (headLength := len(head := data["ParameterNames"])) > 0:
+                    if len(data["Records"]) > 0:
+                        callback = []
+                        for job in data["Records"]:
+                            callback.append({head[j]: job[j] for j in range(headLength)})
+                        timestamp = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M [UTC]")
+                        if "Extras" in data:
+                            return {
+                                "success": "true",
+                                "result": callback,
+                                "total": total,
+                                "extra": data["Extras"],
+                                "date": timestamp,
+                            }
+                        return {"success": "true", "result": callback, "total": total, "date": timestamp}
+                    return {"success": "false", "result": "", "error": "There are no data to display"}
+                return {"success": "false", "result": "", "error": "ParameterNames field is undefined"}
+            return {"success": "false", "result": "", "error": "Data structure is corrupted"}
+        return {"success": "false", "result": "", "error": "There were no data matching your selection"}
```

Comment:
I'm ignoring most of style comments but I think this function is bad enough that we should fix it:

```suggestion
        total = len(data.get("TotalRecords", []))
        if total == 0:
            return {"success": "false", "result": "", "error": "There were no data matching your selection"}
        if not ("ParameterNames" in data and "Records" in data):
            return {"success": "false", "result": "", "error": "Data structure is corrupted"}
        head = data.get("ParameterNames", [])
        if len(head) == 0:
            return {"success": "false", "result": "", "error": "ParameterNames field is undefined"}
        if len(data["Records"]) == 0:
            return {"success": "false", "result": "", "error": "There are no data to display"}
        callback = []
        for job in data["Records"]:
            callback.append(dict(zip(head, job)))
        timestamp = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M [UTC]")
        if "Extras" in data:
            return {
                "success": "true",
                "result": callback,
                "total": total,
                "extra": data["Extras"],
                "date": timestamp,
            }
        return {"success": "true", "result": callback, "total": total, "date": timestamp}     
```

Reply:
Also I think we should be a little more selective with the use of the `:=` operator, things like this are quite tricky to parse:

```
if (headLength := len(head := data["ParameterNames"])) > 0:
```